.buildkite/scripts/build-docs.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -euo pipefail
+set +x
+
+# Configure the git author and committer information
+export GIT_AUTHOR_NAME='Buildkite CI'
+export GIT_AUTHOR_EMAIL='buildkite@elasticsearch-ci.elastic.co'
+export GIT_COMMITTER_NAME=$GIT_AUTHOR_NAME
+export GIT_COMMITTER_EMAIL=$GIT_AUTHOR_EMAIL
+
+
+build_args=""
+rebuild_opt=""
+broken_links_opt=""
+
+# From https://github.com/elastic/infra/blob/master/ci/jjb/elasticsearch-ci/defs/elastic-docs/pull-requests.yml#L110
+# Per https://github.com/elastic/docs/issues/1821, always rebuild all
+# books for PRs to the docs repo, for now.
+# When https://github.com/elastic/docs/issues/1823 is fixed, this
+# should be removed and the original behavior restored.
+#rebuild_opt="--rebuild"
+
+if [[ "${REBUILD}" == 'true' ]]; then
+  rebuild_opt="--rebuild"
+fi
+
+if [[ "${BROKEN_LINKS}" == 'skip' ]]; then
+  broken_links_opt="--skiplinkcheck"
+fi
+
+if [[ "${BROKEN_LINKS}" == 'warn' ]]; then
+  broken_links_opt="--warnlinkcheck"
+fi
+
+# When running on a branch or on main
+if [[ "${BUILDKITE_PULL_REQUEST}" == "false" ]]; then
```

Comment:
Does this run when you push to a branch but don't create a PR? Or does the `branches: "master"` [line](https://github.com/elastic/docs/pull/2774/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53R33)  in `pipeline.yml` prevent this?

We only want to use this for the `master` branch during the migration, and not have other builds fighting over this branch.

Reply:
the branches: "master" line prevents it